src/react/ImgFile.ts

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect, ImgHTMLAttributes } from "react";
+import { DocFileMeta } from "use-fireproof";
+
+const { URL } = window;
+
+// Union type to support both direct File objects and metadata objects
+type FileType = File | DocFileMeta;
+
+interface ImgFileProps extends Omit<ImgHTMLAttributes<HTMLImageElement>, "src"> {
+  file?: FileType;
+  /**
+   * @deprecated Use 'file' instead. This is for internal use only to support legacy code.
+   * @internal
+   */
+  meta?: FileType;
+}
+
+export function ImgFile({ file, meta, ...imgProps }: ImgFileProps) {
+  const [imgDataUrl, setImgDataUrl] = useState("");
+
+  // Use meta as fallback if file is not provided (for backward compatibility)
+  const fileData = file || meta;
+
+  useEffect(() => {
+    if (!fileData) return;
+
+    // Helper function to determine if the object is a File
+    const isFile = (obj: FileType): obj is File => obj instanceof File;
```

Comment:
never ever use instanceof ----> try to poke in the type like 
"method1" in File